### DISCUSSION TITLE
Point values and manipulation in egui::plot::Points

### DISCUSSION THREAD

**Discussion by resonant-jovian** [emilk/egui#856] [Q&A]
Hey,

I am making a system where the user is given some points that I want them to be able to move around. I then want to take the x and y values of all of these points and put them into matrices. I am then going to take these points and do some math with them (derivation, integration, and least square method) and return some constants for a 4th level polynomial. I then want to display all these points and that function on a 2D graph.

Excluding the math stuff, this is the egui code I have so far. The problems I have are point manipulation and getting the x and y values to send to my math function.

```
        let point1 = Points::new(Values::from_values(vec![Value::new(0., 0.)]));
        let point2 = Points::new(Values::from_values(vec![Value::new(1., 0.)]));
        let point3 = Points::new(Values::from_values(vec![Value::new(2., 0.)]));
        let point4 = Points::new(Values::from_values(vec![Value::new(3., 0.)]));
        let point5 = Points::new(Values::from_values(vec![Value::new(4., 0.)]));
        let x_m = [[point1.x, point2.x, point3.x, point4.x, point5.x]];
        let y_m = [[point1.y, point2.y, point3.y, point4.y, point5.y]];
        let x_4 = acceleration(x_m, y_m).0; //a
        let x_3 = acceleration(x_m, y_m).1; //b
        let x_2 = acceleration(x_m, y_m).2; //c
        let x_1 = acceleration(x_m, y_m).3; //d
        let x_0 = acceleration(x_m, y_m).4; //e
        let function = ((point1.x)..(point5.x)).map(|i| {
            let x = i as f64 * 0.01;
            Value::new(
                x,
                x_4 * x.powf(4.) + x_3 * x.powf(3.) + x_2 * x.powf(2.) + x_1 * x + x_0,
            )
        });
        let line = Line::new(Values::from_values_iter(function));
        ui.add(
            Plot::new("my_plot")
                .line(line)
                .points(point1)
                .points(point2)
                .points(point3)
                .points(point4)
                .points(point5)
                .view_aspect(2.0),
        );

```

Thanks in advance!
/Albin Sjögren

**Comment by emilk** [comment]
Being able to manipulate plot items from within a plot is actively being worked on in https://github.com/emilk/egui/pull/766 and https://github.com/emilk/egui/issues/734 by @EmbersArc

**Reply by resonant-jovian** [reply]
Okay and getting values from points?

**Reply by emilk** [reply]
Converting between screen coordinates and plot coordinates is part of that work, yes.

**Reply by resonant-jovian** [reply]
Great
